src/config/configJson.ts

```diff
@@ -37,14 +43,10 @@ export default class ConfigJsonConfigBackend {
 		await Bluebird.using(this.writeLockConfigJson(), async () => {
 			let changed = false;
 			_.forOwn(keyVals, (value, key: T) => {
-				if (this.cache[key] !== value) {
+				if (this.schema[key] != null && !_.isEqual(this.cache[key], value)) {
```

Comment:
Could you comment on the reason to move this clause here?

```
this.schema[key] != null
```

Reply:
This offers better protection against changing values in config.json that aren't part of the Supervisor's config.json schema. If the non-null schema check is in the inner `if` branch, there is the possibility of changing the value for a key (without deleting it) even if the key is not present in the config.json schema. 

Although, technically, the type signature for the class setter already restricts possible keys to those in the schema. Should we remove this check altogether?

Reply:
> Although, it's also fine if we remove this check, since the type signature for the class setter already restricts possible keys to those in the schema. Should we remove this check altogether?

I think let's keep it for now. I don't trust the codebase to not be using `as any` or failing to check for nulls on an array element (see [strictNullChecks](https://www.typescriptlang.org/tsconfig/#strictNullChecks))